src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs

```diff
@@ -54,12 +54,12 @@ public class NoPaginationRequest : IPaginationRequest
 
         public int TotalPages(int totalItemSize) => 1;
 
-        public PagedQueryable<T> Paginate<T>(SortedQueryable<T> source, Func<IQueryable<T>, int> countFunc = null)
+        public PagedQueryable<T> Paginate<T>(SortedQueryable<T> source, Func<IQueryable<T>, long> countFunc = null)
         {
             var items = source.Items;
-            var limit = countFunc?.Invoke(items) ?? items.Count();
+            var limit = countFunc?.Invoke(items) ?? items.LongCount();
 
-            var paginationInfo = new PaginationInfo(0, limit, limit, 1);
+            var paginationInfo = new PaginationInfo(0, (int) limit, limit, 1);
```

Comment:
Why not change `PaginationInfo` to accept long long?

Reply:
Limit (second parameter for paginationInfo) is not the same as Total Items.

Reply:
Ok, but if total items is a long, it is possible a person will enter a limit which exceeds an int too. So we need to change Limit to long too, no?

Reply:
`Limit` = `Pagesize`. Which is currently controlled by us with a maximum.
I would not encourage to have a pagesize larger than an `int`.

Reply:
Right.... Mistaken it with offset :p